starknet_py/tests/e2e/account/account_test.py

```diff
@@ -274,7 +275,7 @@ async def test_deploy_account_raises_on_no_enough_funds(deploy_account_details_f
                 class_hash=class_hash,
                 salt=salt,
                 key_pair=key_pair,
-                client=GatewayClient(net="testnet"),
+                client=GatewayClient(net=Network("testnet")),
```

Comment:
It's a huge breaking change, as practically almost every use case uses client somehow.
Deprecations would for sure be needed.

Also I'm not a fan of creating a new network every time a client is constructed. Maybe we could create Network objects of `testnet`, `testnet2` and `mainnet` somewhere, and just import them, the same way it was done previously. However it would still require users to change the way they use clients currently.

Another, less breaking approach, would be not to change the constructor, and create Network internally from the predefined networks.